build.gradle

```diff
@@ -152,3 +151,40 @@ task generate() {
   // injecting themselves as a task dependency into this task.  E.g., via
   // `afterEvaluate.rootProject.generate.dependsOn(generateProto)`
 }
+
+// Prompt the user for a publication passsword to sign archives or upload artifacts, if requested
```

Comment:
The `askpass` behavior needs to be in root build.gradle becuase not all projects have access to `publish.gradle`, right?

Reply:
That was left in the root `build.gradle` because that's where it had effectively been before.

Old style: publication was shoved into a `subprojects {` configuration, and `askpass` was outside that configuration, implicitly applied to the root project that called the `apply from: 'publish.gradle'`.

New style: publication is explicitly requested in the relevant subprojects via their own `apply from: 'publish.gradle'`.  The `askpass` logic which had been applied at root is now explicitly owned by the root's `build.gradle`.

To my eye, that logic is just a sanity check that gets run once the graph is ready.  That seems like a thing to be owned by root.